utils/fp.ts

```diff
@@ -0,0 +1,10 @@
+export const withValue = <T, R>(
```

Comment:
@3mcd I stole this from your branch because I just couldn't wait :D

Reply:
Sweet, glad its useful!!